### PR TITLE
#164797110 setup social login with google, facebook and twitter

### PR DESCRIPTION
**Pull request by igbominadeveloper**
#### What does this PR do?
 setup social login with google
#### Description of Task to be completed?
- [x] setup social login for google accounts
- [x] setup social login for twitter accounts
- [x] setup social login for facebook accounts
#### Any background context you want to provide?
Users with Google accounts should be able to login with their accounts to ease the signup/login process on Author's Haven
#### What are the relevant pivotal tracker stories?(if applicable)
[164797110](https://www.pivotaltracker.com/story/show/164797110)
#### Screenshots
Screenshot 2019-05-15 at 4 25 35 PM
Screenshot 2019-05-15 at 4 25 38 PM
Screenshot 2019-05-15 at 4 25 45 PM

### CODE REVIEW

File: src/utils/testHelpers.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { initialState } from '../store/modules/auth';
+
+export const findByTestAttribute = (component, attribute) => {
+  const wrapper = component.find(`[data-test='${attribute}']`);
+  return wrapper;
+};
+
+export const getLocation = () => ({
```
Comment (Japhethca) [inline src/utils/testHelpers.js]: we can move this back to the test files.

File: src/utils/testHelpers.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  search:
+    '?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjI3MGEzN2Y3LTc2NWEtNDYyNi04MzE3LTI5ZTZhNjJkOTdiOSIsImVtYWlsIjoiZmF2b3VyYWZvbGF5YW5AZ21haWwuY29tIiwidXNlcm5hbWUiOiJmYXZvdXJhZm9sYXlhbkBnbWFpbC5jb20iLCJpYXQiOjE1NTc5ODA2NjAsImV4cCI6MTU2MDU3MjY2MH0.WUI2HrAEelHEWobpIV9_saSeZghfdC8tUwfwrHTEStI&isNewRecord=false',
+});
+
+const mockStore = configureStore([thunk]);
+export const setupStore = () => mockStore(initialState);
+
+export const signupMockData = {
```
Comment (Japhethca) [inline src/utils/testHelpers.js]: same as above

File: src/store/modules/auth.js [hunk 3]
```diff
@@ -85,6 +96,43 @@ export const signupUser = userData => {
   };
 };
 
+export const socialAuth = (history, location) => {
+  return dispatch => {
+    try {
+      dispatch(loginInitialize());
+      let query = location.search;
```
Comment (Japhethca) [inline src/store/modules/auth.js]: We can use `qs` package to handle query string parsing.
https://www.npmjs.com/package/qs

File: src/utils/testHelpers.js (hunk 2 above)
Comment (igbominadeveloper) [reply src/utils/testHelpers.js]: I have moved it

File: src/utils/testHelpers.js (hunk 1 above)
Comment (igbominadeveloper) [reply src/utils/testHelpers.js]: Done

File: src/store/modules/auth.js (hunk 3 above)
Comment (igbominadeveloper) [reply src/store/modules/auth.js]: Done.

File: src/components/presentationals/SocialAuth/SocialAuth.test.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  const props = {
+    history: {
+      push: jest.fn(),
+    },
+    location,
+  };
+
+  it.only('Mounts correctly without errors', () => {
```
Comment (Japhethca) [inline src/components/presentationals/SocialAuth/SocialAuth.test.jsx]: let's remove `only` from the test case.

**Review by Japhethca — APPROVED** [review]

**Pull request merged**